test/lib/agent_helper.js

```diff
@@ -71,639 +65,572 @@ function FakeSegment(transaction, duration, name = 'FakeSegment') {
+ */
+helper.runInTransaction = (agent, type, callback) => {
+  if (!callback && typeof type === 'function') {
+    callback = type
+    type = undefined
+  }
+  if (!(agent && callback)) {
+    throw new TypeError('Must include both agent and function!')
+  }
+  type = type || 'web'
+
+  // if the agent hasn't been started, set to a state that can collect transactions.
+  // do not override states for an agent that is already started or in the
+  // process of starting.
+  if (agent._state === 'stopped') {
+    agent.setState('started')
+  }
+
+  return agent.tracer.transactionNestProxy(type, () => {
+    const transaction = agent.getTransaction()
+    return callback(transaction)
+  })() // <-- invoke immediately
+}
+
+/**
+ * Proxy for runInTransaction that names the transaction that the
+ * callback is executed in
+ */
+helper.runInNamedTransaction = (agent, type, callback) => {
+  if (!callback && typeof type === 'function') {
+    callback = type
+    type = undefined
+  }
+
+  return helper.runInTransaction(agent, type, (transaction) => {
+    transaction.name = 'TestTransaction'
+    return callback(transaction)
+  })
+}
+
+helper.runInSegment = (agent, name, callback) => {
+  const tracer = agent.tracer
+
+  return tracer.addSegment(name, null, null, null, callback)
+}
+
+/**
+ * Stub to bootstrap a memcached instance
+ *
+ * @param {Function} callback The operations to be performed while the server
+ *                            is running.
+ */
+helper.bootstrapMemcached = (callback) => {
+  const Memcached = require('memcached')
+  const memcached = new Memcached(params.memcached_host + ':' + params.memcached_port)
+  memcached.flush((err) => {
+    memcached.end()
+    callback(err)
+  })
+}
+
+/**
+ * Select Redis DB index and flush entries in it.
+ *
+ * @param {redis} [redis]
+ * @param {number} dbIndex
+ * @param {function} callback
+ *  The operations to be performed while the server is running.
+ */
+helper.flushRedisDb = (client, dbIndex) => {
+  return new Promise((resolve, reject) => {
+    client.select(dbIndex, (err) => {
+      if (err) {
+        client.end(true)
+        reject(err)
+      }
 
-      res.on('end', () => {
-        if (typeof callback === 'function') {
-          const body = contentType?.includes('application/json') ? JSON.parse(rawData) : rawData
-          // assign body to res as when this method is promisified it can only return 2 args: err, result
-          res.body = body
-          callback(null, res, body)
+      client.flushdb((err) => {
+        if (err) {
+          reject(err)
         }
+
+        resolve()
+      })
+    })
+  })
+}
+
+helper.withSSL = () => {
+  return Promise.all([fs.readFile(KEYPATH), fs.readFile(CERTPATH), fs.readFile(CAPATH)])
+}
+
+helper.randomPort = (callback) => {
+  const net = require('net')
+  // Min port: 1024 (without root)
+  // Max port: 65535
+  // Our range: 1024-65024
+  const port = Math.ceil(Math.random() * 64000 + 1024)
```

Comment:
For what it's worth, this could be:

```suggestion
  const port = require('crypto').randomInt(1024, 65024)
```

Reply:
I'd like to stick with this kind of refactor in a diff PR, I agree though. I'll queue that one up later